test/functional/custom-transform.spec.ts

```diff
@@ -66,62 +125,51 @@ describe("custom transformation decorator", () => {
 
     });
 
-    it("versions and groups should work with @Transform decorator too", () => {
+    it("Non class objects should not break if transform is used", () => {
```

Comment:
looks like issue either in how you use routing-controller, either there is a bug in routing-controllers. I'll need to take a look on how you use routing-controllers first, best is to have a repo with your issue reproduction